server/src/main/java/io/seata/server/session/SessionHelper.java

```diff
@@ -97,7 +97,7 @@ public static void endCommitted(GlobalSession globalSession) throws TransactionE
      * @throws TransactionException the transaction exception
      */
     public static void endCommitFailed(GlobalSession globalSession) throws TransactionException {
-        globalSession.changeStatus(GlobalStatus.CommitFailed);
+        globalSession.setStatus(GlobalStatus.CommitFailed);
```

Comment:
What will happen ,if TC Server goes down between `setStatus` and `end` ?

Reply:
Will retry.